test/ng/directive/ngOptionsSpec.js

```diff
@@ -743,6 +743,37 @@ describe('ngOptions', function() {
     expect(options.eq(3)).toEqualOption('c');
   });
 
+  it('should remove the "selected" attribute from the previous option when the model changes', function() {
+    scope.values = [{id: 10, label: 'ten'}, {id:20, label: 'twenty'}];
+
+    createSelect({
+      'ng-model': 'selected',
+      'ng-options': 'item.label for item in values'
+    }, true);
+
+    var resetButton = angular.element('<input type="reset" />');
+    formElement.append(resetButton);
+
+    scope.selected = scope.values[0];
+    scope.$digest();
+
+    var options = element.find('option');
+    expect(options[1].getAttribute('selected')).toBe('selected');
```

Comment:
It's based on personal experience. I tried a fix for this myself and I missed the empty option the first time :smile: